src/backend/distributed/operations/auto_shard_split.c

```diff
@@ -52,90 +51,110 @@
 #include "utils/guc_tables.h"
 #include "distributed/commands/utility_hook.h"
 
-#define MAX_SHARD_SIZE  1
+
+#define MAX_SHARD_SIZE  20000
 PG_FUNCTION_INFO_V1(citus_auto_shard_split_start); 
 
-void Citus_Table_List(void)
-{
-	Assert(CitusHasBeenLoaded() && CheckCitusVersion(WARNING));
+const char* GetSplitQuery(int64 shardminvalue , int64 shardmaxvalue , int64 nodeid , int64 shardid){
 
-	/* first, we need to iterate over pg_dist_partition */
-	List *citusTableIdList = CitusTableTypeIdList(DISTRIBUTED_TABLE);
+    char* query = NULL;
+    int64 query_length =0;
+
+    int64 midpoint = (shardminvalue + ((shardmaxvalue-shardminvalue)>>1));
+    query_length = snprintf(NULL, 0 , "SELECT citus_split_shard_by_split_points(%ld, ARRAY['%ld'], ARRAY[%ld,%ld], 'block_writes')",
```

Comment:
use appendString ti form query.
snprintf, calloc etc should not be used.

Reply:
appendStringInfo used